applications/StructuralMechanicsApplication/custom_response_functions/structural_response.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import print_function, absolute_import, division
+
+# importing the Kratos Library
+from KratosMultiphysics import *
+import structural_mechanics_analysis
+
+import time as timer
+
+class ResponseFunctionBase(object):
+    def __init__(self, identifier, project_parameters):
+        self.identifier = identifier
+        self.project_parameters = project_parameters
+    def Initialize(self):
+        pass
+    def CalculateValue(self):
+        raise NotImplementedError("CalculateValue needs to be implemented by the base class")
+    def CalculateGradient(self):
+        raise NotImplementedError("CalculateGradient needs to be implemented by the base class")
+    def GetShapeGradient(self):
+        raise NotImplementedError("GetShapeGradient needs to be implemented by the base class")
+    def Finalize(self):
+        pass
+
+class SimpleResponseFunctionWrapper(ResponseFunctionBase):
+    def __init__(self, identifier, project_parameters, response_function_utility, model_part = None):
+        super(SimpleResponseFunctionWrapper, self).__init__(identifier, project_parameters)
+
+        # Create the primal solver
+        with open(project_parameters["primal_settings"].GetString()) as parameters_file:
+            ProjectParametersPrimal = Parameters( parameters_file.read() )
+        self.primal_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(ProjectParametersPrimal, model_part)
+
+        self.primal_analysis.GetModelPart().AddNodalSolutionStepVariable(SHAPE_SENSITIVITY)
+
+        self.response_function_utility = response_function_utility
+    def Initialize(self):
+        self.primal_analysis.Initialize()
+        self.response_function_utility.Initialize()
+    def CalculateValue(self):
+        Logger.PrintInfo("\n> Starting primal analysis for response:", self.identifier)
+        startTime = timer.time()
+        self.primal_analysis.InitializeTimeStep()
+        self.primal_analysis.SolveTimeStep()
+        self.primal_analysis.FinalizeTimeStep()
```

Comment:
Imo, the code style of this file should be improved a bit. So introduce whitespaces and improve the line grouping into code blocks.